help/tquery.md

```diff
@@ -0,0 +1,52 @@
+# Tquery
+
+### Typy danych i operatory filtrów
+
+| typ          | null(5) | = | in   | cmp | like | select | sort |
+|--------------|---------|---|------|-----|------|--------|------|
+| bool         | *       | * | .    | .   | .    | *      | +    |
+| date         | *       | * | *    | *   | .    | *      | +    |
+| datetime     | *       | . | .    | *   | .    | *      | +    |
+| int          | *       | * | *    | *   | *    | *      | +    |
+| string       | *       | * | *    | *   | *    | *      | +    |
+| uuid         | *       | * | *    | .   | .    | *      | .    |
+| text         | *       | . | .    | .   | *    | *(8)   | .    |
+| --------(9)  |         |   |      |     |      |        |      |
+| dict(1)      | *       | * | *    | .   | .    | *      | *    |
+| object(2)    | *       | . | .    | .   | .    | *      | .    |
+| dict_list(3) | *       | * | *(7) | .   | .    | *      | .    |
+| list(4)      | *       | . | .    | .   | .    | *      | .    |
+
+- `null` - operator poprawny tylko dla typów nullable
+- `=` - porównanie z wartością tego samego typu, int-int, string-string, itp., brak możliwości porównania z null ani z
+  pustym stringiem
+- `==`(?) - porównanie typu binary, bardziej kosztowne dla sql'a, ale z zachowaniem wielkości liter, polskich liter,
```

Comment:
Dlaczego porównanie dokładne jest niby bardziej kosztowne niż porównanie mniej-więcej? W ogóle widzę że jest jakieś collate którym można to zrobić, a nie tylko binary.

Reply:
A w ogóle to ten `==` ma sens tylko dla string, a nie dla wszystkich typów które mają `*` w kolumne `=`. Można by to jakoś oznaczyć jednak dla ścisłości. Podobnie regexp jest tylko dla string i text, więc też powinien mieć swoją kolumnę.

Reply:
(?) oznacza, że to jest opis na mniej-więcej, ale po prostu fe ma używać = jeżeli nie ma jakiejś szczególnej potrzeby